controller/sync_strats/rtx.go

```diff
@@ -0,0 +1,156 @@
+/*
+	Copyright NetFoundry, Inc.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+	https://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+package sync_strats
+
+import (
+	"github.com/michaelquigley/pfxlog"
+	"github.com/openziti/edge/controller/env"
+	"github.com/openziti/edge/controller/model"
+	"github.com/openziti/edge/eid"
+	"github.com/openziti/fabric/controller/network"
+	"github.com/openziti/foundation/channel2"
+	"github.com/openziti/foundation/util/concurrenz"
+	"github.com/sirupsen/logrus"
+	"sync"
+)
+
+// routerTx represents a connection from an Edge Router to the controller. Used
+// to asynchronously buffer and send messages to an Edge Router via Start() then Send()
+type routerTx struct {
+	Id         string
```

Comment:
This id is meant to be used to track what sync is currently being worked on. The id is monotonic so they can be compared for "who is newer". This id should be output in log messages to track a sync attempt.